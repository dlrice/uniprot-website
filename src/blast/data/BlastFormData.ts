export type BlastFormValue = {
  fieldName: string;
  selected?: string | undefined;
  selectedLabel?: string | undefined;
  type: BlastFieldTypes;
  values?: { label?: string; value?: string | number }[];
};

export enum BlastFields {
  program = 'Program',
  stype = 'Sequence type',
  email = 'Email',
  sequence = 'Sequence',
  taxon = 'Taxon',
  targetDb = 'Target database',
  threshold = 'E-Threshold',
  matrix = 'Matrix',
  filter = 'Filter',
  gapped = 'Gapped',
  hits = 'Hits',
}

export enum BlastFieldTypes {
  textarea,
  select,
  autocomplete,
}

export type BlastFormValues = { [x in BlastFields]: BlastFormValue };

export default {
  [BlastFields.program]: {
    fieldName: 'program',
    selected: 'blastp',
  },
  [BlastFields.stype]: {
    fieldName: 'stype',
    selected: 'protein',
  },
  [BlastFields.email]: {
    fieldName: 'email',
    selected: 'uuw_dev@uniprot.org',
  },
  [BlastFields.sequence]: {
    fieldName: 'sequence',
    type: BlastFieldTypes.textarea,
  },
  [BlastFields.targetDb]: {
    fieldName: 'database',
    type: BlastFieldTypes.select,
    selected: 'uniprotkb_refprotswissprot',
    values: [
      {
        value: 'uniprotkb_refprotswissprot',
        label: 'UniProtKb',
      },
    ],
  },
  [BlastFields.taxon]: {
    fieldName: 'taxids',
    type: BlastFieldTypes.autocomplete,
  },
  [BlastFields.threshold]: {
    fieldName: 'threshold',
    type: BlastFieldTypes.select,
    selected: '10',
    values: [
      { value: '0.0001' },
      { value: '0.001' },
      { value: '0.01' },
      { value: '0.1' },
      { value: '1' },
      { value: '10' },
      { value: '100' },
      { value: '1000' },
    ],
  },
  [BlastFields.matrix]: {
    fieldName: 'matrix',
    type: BlastFieldTypes.select,
    selected: null,
    values: [
      {
        value: '',
        label: 'Auto',
      },
      { value: '45' },
      { value: '62' },
      { value: '80' },
      { value: '70' },
      { value: '30' },
    ],
  },
  [BlastFields.filter]: {
    fieldName: 'filter',
    type: BlastFieldTypes.select,
    selected: 'F',
    values: [
      { value: 'F', label: 'None' },
      { valueNone: 'T', label: 'Filter low complexity regions' },
      { value: 'mask', label: 'Mask lookup table only' },
    ],
  },
  [BlastFields.gapped]: {
    fieldName: 'gapped',
    type: BlastFieldTypes.select,
    selected: 'true',
    values: [
      { value: 'true', label: 'yes' },
      { value: 'false', label: 'no' },
    ],
  },
  [BlastFields.hits]: {
    fieldName: 'hits',
    type: BlastFieldTypes.select,
    selected: '250',
    values: [
      { value: '50' },
      { value: '100' },
      { value: '250' },
      { value: '500' },
      { value: '750' },
      { value: '1000' },
    ],
  },
};
