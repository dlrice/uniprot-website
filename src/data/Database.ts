import { Database, DatabaseCategory } from '../model/types/databaseTypes';
import EntrySection from '../model/types/EntrySection';

export const databaseCategoryToString = new Map<DatabaseCategory, string>([
  [DatabaseCategory.SEQUENCE, 'Sequence databases'],
  [DatabaseCategory.STRUCTURE, '3D structure databases'],
  [DatabaseCategory.INTERACTION, 'Protein-protein interaction databases'],
  [DatabaseCategory.CHEMISTRY, 'Chemistry'],
  [DatabaseCategory.FAMILY, 'Protein family/group databases'],
  [DatabaseCategory.PTM, 'PTM databases'],
  [DatabaseCategory.POLYMORPHISM, 'Polymorphism and mutation databases'],
  [DatabaseCategory.GEL, '2D gel databases'],
  [DatabaseCategory.PROTEOMIC, 'Proteomic databases'],
  [DatabaseCategory.PROTOCOL, 'Protocols and materials databases'],
  [DatabaseCategory.GENOME, 'Genome annotation databases'],
  [DatabaseCategory.ORGANISM, 'Organism-specific databases'],
  [DatabaseCategory.PHYLOGENOMIC, 'Phylogenomic databases'],
  [DatabaseCategory.PATHWAY, 'Enzyme and pathway databases'],
  [DatabaseCategory.EXPRESSION, 'Gene expression databases'],
  [DatabaseCategory.DOMAIN, 'Family and domain databases'],
  [DatabaseCategory.OTHER, 'Other'],
]);

const databaseCategoryToDatabases = new Map<DatabaseCategory, Database[]>();
databaseCategoryToDatabases.set(DatabaseCategory.SEQUENCE, [
  // Not found: EMBL/GenBank/DDBJ
  Database.EMBL,
  Database.CCDS,
  Database.PIR,
  Database.RefSeq,
  Database.UniGene,
]);
databaseCategoryToDatabases.set(DatabaseCategory.STRUCTURE, [
  Database.PDB,
  Database.SMR,
  Database.DisProt,
  Database.ProteinModelPortal,
]);
databaseCategoryToDatabases.set(DatabaseCategory.INTERACTION, [
  Database.BioGrid,
  Database.DIP,
  Database.IntAct,
  Database.MINT,
  Database.STRING,
]);
databaseCategoryToDatabases.set(DatabaseCategory.CHEMISTRY, [
  Database.BindingDB,
  Database.ChEMBL,
  Database.DrugBank,
  Database.GuidetoPHARMACOLOGY,
]);
databaseCategoryToDatabases.set(DatabaseCategory.FAMILY, [
  // Not found: Database.PptaseDB,
  Database.Allergome,
  Database.CAZy,
  Database.MEROPS,
  Database.MoonProt,
  Database.mycoCLAP,
  Database.PeroxiBase,
  Database.REBASE,
  Database.TCDB,
]);
databaseCategoryToDatabases.set(DatabaseCategory.PTM, [
  // Not found: DEPOD, PhosphoSite, PhosSite
  Database.DEPOD,
  Database.PhosphoSitePlus,
  Database.UniCarbKB,
]);
databaseCategoryToDatabases.set(DatabaseCategory.POLYMORPHISM, [
  Database.dbSNP,
  Database.DMDM,
]);
databaseCategoryToDatabases.set(DatabaseCategory.GEL, [
  Database.SWISS_2DPAGE,
  Database.COMPLUYEAST_2DPAGE,
  Database.DOSAC_COBS_2DPAGE,
  Database.OGP,
  Database.REPRODUCTION_2DPAGE,
  Database.UCD_2DPAGE,
  Database.World_2DPAGE,
]);
databaseCategoryToDatabases.set(DatabaseCategory.PROTEOMIC, [
  Database.MaxQB,
  Database.PaxDb,
  Database.PeptideAtlas,
  Database.PRIDE,
  Database.ProMEX,
]);
databaseCategoryToDatabases.set(DatabaseCategory.PROTOCOL, [Database.DNASU]);
databaseCategoryToDatabases.set(DatabaseCategory.GENOME, [
  Database.Ensembl,
  Database.EnsemblBacteria,
  Database.EnsemblFungi,
  Database.EnsemblMetazoa,
  Database.EnsemblPlants,
  Database.EnsemblProtists,
  Database.GeneCards,
  Database.GeneID,
  Database.KEGG,
  Database.PATRIC,
  Database.UCSC,
  Database.VectorBase,
]);
databaseCategoryToDatabases.set(DatabaseCategory.ORGANISM, [
  Database.ArachnoServer,
  Database.CGD,
  Database.ConoServer,
  Database.CTD,
  Database.dictyBase,
  Database.EchoBASE,
  Database.EcoGene,
  Database.euHCVdb,
  Database.EuPathDB,
  Database.FlyBase,
  Database.GeneReviews,
  Database.Gramene,
  Database.H_InvDB,
  Database.HGNC,
  Database.HPA,
  Database.LegioList,
  Database.Leproma,
  Database.MaizeGDB,
  Database.MIM,
  Database.MGI,
  Database.neXtProt,
  Database.Orphanet,
  Database.PharmGKB,
  Database.PomBase,
  Database.PseudoCAP,
  Database.RGD,
  Database.SGD,
  Database.TAIR,
  Database.TubercuList,
  Database.WormBase,
  Database.Xenbase,
  Database.ZFIN,
]);
databaseCategoryToDatabases.set(DatabaseCategory.PHYLOGENOMIC, [
  Database.eggNOG,
  Database.GeneTree,
  Database.HOGENOM,
  Database.HOVERGEN,
  Database.InParanoid,
  Database.KO,
  Database.OMA,
  Database.OrthoDB,
  Database.PhylomeDB,
  Database.TreeFam,
]);
databaseCategoryToDatabases.set(DatabaseCategory.PATHWAY, [
  Database.BioCyc,
  Database.BRENDA,
  Database.Reactome,
  Database.SABIO_RK,
  Database.SignaLink,
  Database.UniPathway,
]);
databaseCategoryToDatabases.set(DatabaseCategory.EXPRESSION, [
  Database.Bgee,
  Database.ExpressionAtlas,
]);
databaseCategoryToDatabases.set(DatabaseCategory.DOMAIN, [
  Database.HAMAP,
  Database.InterPro,
  Database.Gene3D,
  Database.PANTHER,
  Database.Pfam,
  Database.PIRSF,
  Database.PRINTS,
  Database.ProDom,
  Database.SMART,
  Database.SUPFAM,
  Database.TIGRFAMs,
  Database.PROSITE,
]);
databaseCategoryToDatabases.set(DatabaseCategory.OTHER, [
  // Not found: NextBio
  Database.ChiTaRS,
  Database.EvolutionaryTrace,
  Database.GenomeRNAi,
  Database.GeneWiki,
  Database.PMAP_CutDB,
  Database.PRO,
]);

const entrySectionToDatabaseCategories = new Map<
  EntrySection,
  DatabaseCategory[]
>();
entrySectionToDatabaseCategories.set(EntrySection.Expression, [
  DatabaseCategory.ORGANISM,
  DatabaseCategory.EXPRESSION,
]);
entrySectionToDatabaseCategories.set(EntrySection.FamilyAndDomains, [
  DatabaseCategory.FAMILY,
  DatabaseCategory.PHYLOGENOMIC,
  DatabaseCategory.DOMAIN,
]);
entrySectionToDatabaseCategories.set(EntrySection.Function, [
  DatabaseCategory.PATHWAY,
  DatabaseCategory.FAMILY,
]);
entrySectionToDatabaseCategories.set(EntrySection.Interaction, [
  DatabaseCategory.INTERACTION,
  DatabaseCategory.CHEMISTRY,
]);
entrySectionToDatabaseCategories.set(EntrySection.NamesAndTaxonomy, [
  DatabaseCategory.ORGANISM,
]);
entrySectionToDatabaseCategories.set(EntrySection.PathologyOrBioTech, [
  DatabaseCategory.ORGANISM,
  DatabaseCategory.OTHER,
  DatabaseCategory.FAMILY,
  DatabaseCategory.CHEMISTRY,
  DatabaseCategory.POLYMORPHISM,
]);
entrySectionToDatabaseCategories.set(EntrySection.ProteinProcessing, [
  DatabaseCategory.PROTEOMIC,
  DatabaseCategory.GEL,
  DatabaseCategory.PTM,
  DatabaseCategory.OTHER,
]);
entrySectionToDatabaseCategories.set(EntrySection.Sequence, [
  DatabaseCategory.SEQUENCE,
  DatabaseCategory.GENOME,
  DatabaseCategory.POLYMORPHISM,
]);
entrySectionToDatabaseCategories.set(EntrySection.Structure, [
  DatabaseCategory.STRUCTURE,
  DatabaseCategory.PROTOCOL,
  DatabaseCategory.OTHER,
]);

// Combine multiple maps into single map for quick read access.
export const entrySectionDatabaseToDatabaseCategory = new Map<
  string,
  DatabaseCategory
>();
for (const [
  entrySection,
  databaseCategories,
] of entrySectionToDatabaseCategories) {
  if (databaseCategories) {
    for (const databaseCategory of databaseCategories) {
      const databases = databaseCategoryToDatabases.get(databaseCategory);
      if (databases) {
        for (const database of databases) {
          entrySectionDatabaseToDatabaseCategory.set(
            `${entrySection}_${database}`,
            databaseCategory
          );
        }
      }
    }
  }
}

export default entrySectionDatabaseToDatabaseCategory;
