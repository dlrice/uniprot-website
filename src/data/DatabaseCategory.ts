enum databaseCategory {
  SEQUENCE = 'SEQUENCE',
  STRUCTURE = 'STRUCTURE',
  INTERACTION = 'INTERACTION',
  CHEMISTRY = 'CHEMISTRY',
  FAMILY = 'FAMILY',
  PTM = 'PTM',
  POLYMORPHISM = 'POLYMORPHISM',
  GEL = 'GEL',
  PROTEOMIC = 'PROTEOMIC',
  PROTOCOL = 'PROTOCOL',
  GENOME = 'GENOME',
  ORGANISM = 'ORGANISM',
  PHYLOGENOMIC = 'PHYLOGENOMIC',
  PATHWAY = 'PATHWAY',
  EXPRESSION = 'EXPRESSION',
  DOMAIN = 'DOMAIN',
  OTHER = 'OTHER',
}

export default databaseCategory;
