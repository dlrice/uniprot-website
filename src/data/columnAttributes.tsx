const small = 200; //10rem;
const medium = 400; //15rem;
const large = 600; //20rem;

export type columnAttributes = {
  [columnName: string]: {
    width: number;
    label: string;
  };
};

const columnAttributes: columnAttributes = {
  accession: {
    width: small,
    label: 'Entry',
  },
  id: {
    width: small,
    label: 'Entry Name',
  },
  protein_name: {
    width: large,
    label: 'Protein names',
  },
  gene_names: {
    width: medium,
    label: 'Gene Names',
  },
  organism: {
    width: large,
    label: 'Organism',
  },
};

export default columnAttributes;
