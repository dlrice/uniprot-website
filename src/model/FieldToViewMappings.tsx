import React from 'react';
import SimpleView from './SimpleView';
import { ProteinNames, ProteinNamesData } from './ProteinNames';
import { GeneNames, GeneNamesData } from './GeneNames';
import { Organism, OrganismData } from './Organism';

const FieldToViewMappings: {
  [index: string]: (row: any) => JSX.Element | undefined;
} = {
  accession: (row: { primaryAccession: string }) => (
    <SimpleView
      termValue={row.primaryAccession}
      linkTo={`/uniprotkb/${row.primaryAccession}`}
    />
  ),
  id: (row: { uniProtId: string }) => <SimpleView termValue={row.uniProtId} />,
  protein_name: (data: ProteinNamesData) => {
    return <ProteinNames data={data} />;
  },
  gene_names: (data: GeneNamesData) => {
    return <GeneNames data={data} />;
  },
  organism: (data: OrganismData) => {
    return <Organism data={data} />;
  },
};

export default FieldToViewMappings;
