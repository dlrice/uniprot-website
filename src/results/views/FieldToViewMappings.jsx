import React from 'react';
import get from 'lodash.get';
import SimpleView from './SimpleView';
import NameView from './NameView';

const FieldToViewMappings = {
  accession: row => <SimpleView termValue={row.accession} />,
  id: row => <SimpleView termValue={row.id} />,
  protein_name: (row) => {
    const alternativeNames = [];
    const ecNumber = get(row, 'protein.recommendedName.ecNumber.value');
    if (ecNumber) {
      alternativeNames.push(ecNumber);
    }
    const alternativeName = get(row, 'protein.recommendedName.alternativeName');
    if (alternativeName) {
      alternativeNames.push(alternativeName);
    }
    const props = {
      name: get(row, 'protein.recommendedName.fullName.value'),
      shortName: get(row, 'protein.recommendedName.shortName.value'),
      alternativeNames,
    };
    return <NameView {...props} />;
  },
  gene_name: (row) => {
    const props = {
      name: row.gene ? row.gene[0].name.value : '',
      alternativeNames: [
        // ...row.gene[0].synonyms.map(syn => syn.value),
        // ...row.gene[0].orfNames.map(orf => orf.value),
      ],
    };
    return <NameView {...props} />;
  },
};

export default FieldToViewMappings;
