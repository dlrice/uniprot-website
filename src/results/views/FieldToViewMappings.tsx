import React from 'react';
import idx from 'idx';
import SimpleView from './SimpleView';
import NameView from './NameView';

type ProteinNameRow = {
  protein: {
    recommendedName: {
      ecNumber: {
        value: string;
      };
      fullName: {
        value: string;
      };
      shortName: {
        value: string;
      };
    };
    alternativeName: [{ fullName: { value: string } }];
  };
};

type GeneNameRow = {
  gene: [
    {
      name: {
        value: string;
      };
      synonyms: [{ value: string }];
      orfNames: [{ value: string }];
    }
  ];
};

type OrganismRow = {
  organism: {
    names: [
      {
        type: string;
        value: string;
      }
    ];
  };
};

const organismNameReducer = (type: string) => (
  acc: string,
  organismName: { type: string; value: string },
) => {
  if (organismName.type === type) {
    if (type === 'scientific') {
      return `${acc} ${organismName.value}`;
    }
    return `${acc} (${organismName.value})`;
  }
  return acc;
};

const FieldToViewMappings = {
  accession: (row: { accession: string }) => <SimpleView termValue={row.accession} />,
  id: (row: { id: string }) => <SimpleView termValue={row.id} />,
  protein_name: (row: ProteinNameRow) => {
    const alternativeNames: string[] = [];
    const ecNumber = idx(row, _ => _.protein.recommendedName.ecNumber.value);
    if (ecNumber) {
      alternativeNames.push(ecNumber);
    }
    const alternativeNameArray = idx(row, _ => _.protein.alternativeName);
    if (alternativeNameArray) {
      const alternativeName = alternativeNameArray.map(name => name.fullName.value);
      if (alternativeName.length) {
        alternativeNames.push(...alternativeName);
      }
    }
    const props = {
      name: idx(row, _ => _.protein.recommendedName.fullName.value),
      shortName: idx(row, _ => _.protein.recommendedName.shortName.value),
      alternativeNames,
    };
    return <NameView {...props} />;
  },
  gene_names: (row: GeneNameRow) => {
    const genes = idx(row, _ => _.gene);
    if (!genes || genes.length <= 0) {
      return;
    }
    // const name = genes.map(gene => idx(gene, _ => _.name.value)).join(', ');
    const names: string[] = [];
    const alternativeNames: string[] = [];
    genes.forEach((gene) => {
      if (gene.name) {
        names.push(gene.name.value);
      }
      if (gene.synonyms) {
        alternativeNames.push(...gene.synonyms.map(syn => syn.value));
      }
      if (gene.orfNames) {
        alternativeNames.push(...gene.orfNames.map(orf => orf.value));
      }
    });
    const nameString = names.join(', ');
    const props = { name: nameString, alternativeNames };
    return <NameView {...props} />;
  },
  organism: (row: OrganismRow) => {
    const names = idx(row, _ => _.organism.names);
    if (!names) {
      return;
    }
    const termValue = ['scientific', 'common', 'synonym'].reduce(
      (acc, organismName) => `${acc} ${names.reduce(organismNameReducer(organismName), '')}`,
      '',
    );
    return <SimpleView termValue={termValue} />;
  },
};

export default FieldToViewMappings;
