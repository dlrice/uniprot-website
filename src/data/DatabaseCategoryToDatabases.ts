import DatabaseCategory from './DatabaseCategory';

const databaseCategoryToDatabases = new Map<DatabaseCategory, string[]>([
  [
    DatabaseCategory.SEQUENCE,
    ['EMBL/GenBank/DDBJ', 'CCDS', 'PIR', 'RefSeq', 'UniGene'],
  ],
  [DatabaseCategory.STRUCTURE, ['PDB', 'SMR', 'DisProt', 'ProteinModelPortal']],
  [
    DatabaseCategory.INTERACTION,
    ['BioGrid', 'DIP', 'IntAct', 'MINT', 'STRING'],
  ],
  [
    DatabaseCategory.CHEMISTRY,
    ['BindingDB', 'ChEMBL', 'DrugBank', 'GuidetoPHARMACOLOGY'],
  ],
  [
    DatabaseCategory.FAMILY,
    [
      'Allergome',
      'CAZy',
      'MEROPS',
      'MoonProt',
      'mycoCLAP',
      'PeroxiBase',
      'PptaseDB',
      'REBASE',
      'TCDB',
    ],
  ],
  [DatabaseCategory.PTM, ['DEPOD', 'PhosphoSite', 'PhosSite', 'UniCarbKB']],
  [DatabaseCategory.POLYMORPHISM, ['dbSNP', 'DMDM']],
  [
    DatabaseCategory.GEL,
    [
      'SWISS-2DPAGE',
      'COMPLUYEAST-2DPAGE',
      'DOSAC-COBS-2DPAGE',
      'OGP',
      'REPRODUCTION-2DPAGE',
      'UCD-2DPAGE',
      'World-2DPAGE',
    ],
  ],
  [
    DatabaseCategory.PROTEOMIC,
    ['MaxQB', 'PaxDb', 'PeptideAtlas', 'PRIDE', 'ProMEX'],
  ],
  [DatabaseCategory.PROTOCOL, ['DNASU']],
  [
    DatabaseCategory.GENOME,
    [
      'Ensembl and Ensembl Genomes',
      'Ensembl',
      'EnsemblBacteria',
      'EnsemblFungi',
      'EnsemblMetazoa',
      'EnsemblPlants',
      'EnsemblProtists',
      'GeneCards',
      'GeneID',
      'KEGG',
      'PATRIC',
      'UCSC',
      'VectorBase',
    ],
  ],
  [
    DatabaseCategory.ORGANISM,
    [
      'ArachnoServer',
      'CGD',
      'ConoServer',
      'CTD',
      'dictyBase',
      'EchoBASE',
      'EcoGene',
      'euHCVdb',
      'EuPathDB',
      'FlyBase',
      'GeneReviews',
      'Gramene',
      'H-InvDB',
      'HGNC',
      'HPA',
      'LegioList',
      'Leproma',
      'MaizeGDB',
      'MIM',
      'MGI',
      'neXtProt',
      'Orphanet',
      'PharmGKB',
      'PomBase',
      'PseudoCAP',
      'RGD',
      'SGD',
      'TAIR',
      'TubercuList',
      'WormBase',
      'Xenbase',
      'ZFIN',
    ],
  ],
  [
    DatabaseCategory.PHYLOGENOMIC,
    [
      'eggNOG',
      'GeneTree',
      'HOGENOM',
      'HOVERGEN',
      'InParanoid',
      'KO',
      'OMA',
      'OrthoDB',
      'PhylomeDB',
      'TreeFam',
    ],
  ],
  [
    DatabaseCategory.PATHWAY,
    ['BioCyc', 'BRENDA', 'Reactome', 'SABIO-RK', 'SignaLink', 'UniPathway'],
  ],
  [DatabaseCategory.EXPRESSION, ['Bgee', 'ExpressionAtlas']],
  [
    DatabaseCategory.DOMAIN,
    [
      'HAMAP',
      'InterPro',
      'Gene3D',
      'PANTHER',
      'Pfam',
      'PIRSF',
      'PRINTS',
      'ProDom',
      'SMART',
      'SUPFAM',
      'TIGRFAMs',
      'PROSITE',
    ],
  ],
  [
    DatabaseCategory.OTHER,
    [
      'ChiTaRS',
      'EvolutionaryTrace',
      'GenomeRNAi',
      'GeneWiki',
      'NextBio',
      'PMAP-CutDB',
      'PRO',
    ],
  ],
]);

export default databaseCategoryToDatabases;
