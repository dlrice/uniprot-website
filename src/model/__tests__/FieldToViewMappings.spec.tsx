import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import FieldToViewMappings from '../FieldToViewMappings';
import data from './modelData.json';

configure({ adapter: new Adapter() });

describe('FieldToViewMappings component', () => {
  test('should render accession', () => {
    const wrapper = shallow(FieldToViewMappings.accession(data));
    expect(wrapper.debug()).toMatchSnapshot();
  });

  test('should render id', () => {
    const wrapper = shallow(FieldToViewMappings.id(data));
    expect(wrapper.debug()).toMatchSnapshot();
  });

  test('should render protein_name', () => {
    const wrapper = shallow(FieldToViewMappings.protein_name(data));
    expect(wrapper.debug()).toMatchSnapshot();
  });

  test('should render gene_names', () => {
    const wrapper = shallow(FieldToViewMappings.gene_names(data));
    expect(wrapper.debug()).toMatchSnapshot();
  });

  test('should render organism', () => {
    const wrapper = shallow(FieldToViewMappings.organism(data));
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
