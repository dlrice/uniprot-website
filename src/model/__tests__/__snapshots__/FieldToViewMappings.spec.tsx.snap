// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldToViewMappings component should render accession 1`] = `
"<Link to=\\"/uniprotkb/P21802\\" replace={false}>
  P21802
</Link>"
`;

exports[`FieldToViewMappings component should render gene_names 1`] = `"<NameView name=\\"some Gene\\" alternativeNames={{...}} />"`;

exports[`FieldToViewMappings component should render id 1`] = `
"<span>
  uniprot id
</span>"
`;

exports[`FieldToViewMappings component should render organism 1`] = `"<SimpleView termValue=\\"scientific name (common name)  (synonyms 1)\\" />"`;

exports[`FieldToViewMappings component should render protein_name 1`] = `"<NameView name=\\"rec full Name\\" shortNames={[undefined]} alternativeNames={{...}} />"`;
